src/vulkan/transfer_buffer.h

```diff
@@ -30,10 +31,13 @@ class Device;
 /// Wrapper around a Vulkan VkBuffer object.
 class TransferBuffer : public Resource {
  public:
-  TransferBuffer(Device* device, uint32_t size_in_bytes);
+  TransferBuffer(Device* device,
+                 uint32_t size_in_bytes,
+                 Format* format = nullptr);
```

Comment:
nit: Remove the default value and pass `nullptr` at the various callers. I find default params makes refactoring a lot harder.

Reply:
done